scipy/interpolate/_rgi.py

```diff
@@ -0,0 +1,604 @@
+__all__ = ['RegularGridInterpolator', 'interpn']
+
+import itertools
+
+import numpy as np
+
+from .interpnd import _ndim_coords_from_arrays
+from ._cubic import PchipInterpolator
```

Comment:
It seems that `PchipInterpolator` is not used in this PR. Is this preparation for #16224?

Reply:
Indeed, it's a mistake in splitting the original PR. Ok to keep it like this?